flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableFactory.java

```diff
@@ -277,10 +277,10 @@ private void validateConfigOptions(ReadableConfig config) {
 		final Optional<JdbcDialect> dialect = JdbcDialects.get(jdbcUrl);
 		checkState(dialect.isPresent(), "Cannot handle such jdbc url: " + jdbcUrl);
 
-		if (config.getOptional(USERNAME).isPresent()) {
-			checkState(config.getOptional(PASSWORD).isPresent(),
-				"Database username must be provided when database password is provided");
-		}
+		checkAllOrNone(config, new ConfigOption[]{
```

Comment:
Validation for `URL` and `TABLE_NAME` is not necessary because the find service already does that.